### ISSUE TITLE
How to use NotifyOnProgress

### ISSUE THREAD

**Issue by ggolda** [rosenbjerg/FFMpegCore#277]
Can't figure out how to use `NotifyOnProgress` method that should return percentages and accepts `TimeSpan`.

Could you please provide an example? What `TimeSpan` should I provide? A complete duration of a video file?

I'm trying to build a progress bar and want to use this method to build it.

**Comment by jihadkhawaja** [comment]
@ggolda the TimeSpan is indeed the video duration fetched with **FFProbe.Analyse(inputPath)**

```
Action<double> progressHandler = new Action<double>(p =>
  {
      //progress value from 0 to 100
      Debug.WriteLine(p);
  });

  await FFMpegArguments
   .FromFileInput(inputpath)
   .OutputToFile(outputpath, true, options => options
       .WithVideoCodec(videoCodec)
       .WithConstantRateFactor(21)
       .WithAudioCodec(audioCodec)
       .WithVariableBitrate(4)
       .WithVideoFilters(filterOptions => filterOptions
          .Scale(videoSize))
       .WithFastStart()).NotifyOnProgress(progressHandler, videoDuration)
   .ProcessAsynchronously();
```

**Comment by rosenbjerg** [comment]
I have added some documentation comments to the NotifyOn* methods, to hopefully clarify this for others.
Thank you @jihadkhawaja for providing an answer to @ggolda

**Comment by ggolda** [comment]
Thanks for response!

**Comment by grothag** [comment]
So it reports only final status ? 100% ? Nothing in between ?

**Comment by jihadkhawaja** [comment]
@grothag Sorry can you clarify? it does report progress from 0 to 100 %

**Comment by cvcvideo** [comment]
Hallo, i not know if anybody else has same problem. 
The progressHandler Action only fire on all my tests if progress is at 100%. 
I not get any progress updates or Action event fire during progress timespan .... any idea ? 
Is there anything to set/update in background ? 
Same with onTimeProgress .... All test with .net6 .... possibly it is .net6 error ?

This two voids only called at the end of the progress at 100%:
void OnPercentageProgess(double percentage) => percentageDone = percentage;
void OnTimeProgess(TimeSpan time) => timeDone = time;

Inside FFMpegArguments ( Async ):
.NotifyOnProgress(OnPercentageProgess, analysis.Duration)
.NotifyOnProgress(OnTimeProgess)
.ProcessAsynchronously();

**Comment by rosenbjerg** [comment]
@cvcvideo that issue was tracked in #331 and fixed in #339
